sympy/concrete/summations.py

```diff
@@ -1158,7 +1165,7 @@ def eval_sum_symbolic(f, limits):
     if result is not None:
         n = result[n]
 
-        if n.is_Integer:
+        if n.is_Integer or n.is_real:
```

Comment:
Which problem will this solve? It seems like the requirements are quite different. `is_Integer` will only be true if `n` is an actual integer, i.e., a number. `is_real` will be True for quite a lot of expressions, including symbols that are real, but not necessarily integer.

I'm not saying that it is a bad idea, just a bit surprising. `is_integer` would be less surprising.

Reply:
If n is a symbol or an expression, the next comparison may throw an error and so on. So it seems like maybe this is not needed (as in not tested for the case where n is an unevaluated expression).

Reply:
Reading up a bit more. It seems like n will match the exponent, so I guess this is for sqrt? Then, one should consider if the expressions below are valid for Rational numbers (bernoulli is at least not) and probably test with `n.is_Rational` instead of `n.is_real`. Even though `harmonic(oo, S.Half)` evaluates to `oo`, I am not sure it will be correct for other cases. So probably better to have a test after this to deal with the cases that can be solved explicitly.

You then should add some tests that uses this for sqrt-summations that doesn't evaluate to `oo`.

Reply:
Yeah  exactly what I wanted to fix , `Sum( sqrt(x), (x, 1, oo)).doit()` or rather any non-integer exponent like `Sum( x**(2.5), (x, 1, oo)).doit()` wasn't returning `oo` and the check is for that . Yess I'll be adding all tests for all these , was waiting for initial reviews by you'll !

Reply:
> Reading up a bit more. It seems like n will match the exponent, so I guess this is for sqrt? Then, one should consider if the expressions below are valid for Rational numbers (bernoulli is at least not) and probably test with `n.is_Rational` instead of `n.is_real`. Even though `harmonic(oo, S.Half)` evaluates to `oo`, I am not sure it will be correct for other cases. So probably better to have a test after this to deal with the cases that can be solved explicitly.
> 
> You then should add some tests that uses this for sqrt-summations that doesn't evaluate to `oo`.
Well we won't get anything of this sort with .is_Rational  probably .is_real would be better !
```
>>> Sum(x**2.66, (x, 1, oo)).doit()
Sum(x**2.66, (x, 1, oo))
>>> Sum(1/(x**0.99), (x, 1, oo)).doit()
Sum(x**(-0.99), (x, 1, oo)) 
```

Reply:
Floats are tricky in general. I suggest add `Sum(x**(S(8)/3), (x, 1, oo)).doit()` and, say, `Sum(x**S(9)/10, (x, 1, oo)).doit()` instead.

And for sqrt-test that doesn't sum up to `oo`, say, `Sum(1/sqrt(x), (x, 1, 1000)).doit()` which currently is a 1000 term series, but I assume could be something else. (If it is still a 1000 term series, maybe it is a bad idea to add it...)

Reply:
Nah, I may be a bit confused here. Add what you think are good tests here, but it would be interesting to see a case evaluating the bernoulli and the harmonic lines where n is real and the result is not `oo` or `-oo`. Those are the ones I am a bit concerned about.

(Still, floats are a bit non-intuitive and may lead to weird errors, although they may work here.)

Reply:
yeah even I wasn't completely sure what all tests (specifically those which don't evaluate to `oo`) would be a good fit here ,will give a more thorough thought regarding what to add according to your comments . Yeah truly speaking I didn't consider much when I added `n.is_real` .I'll try some stuff out , if you have something very particular in mind let me know !Thanks

Reply:
Those suggested above would be good. They will trigger the new parts of the code I think (n not being integer).

In addition:
m = Symbol('m', real=True)
Sum(x**m, (x, 1, oo)).doit()
```

Chances are that this will throw an error though, which then indicates that `is_real` is not what you want.
Sum(

Reply:
hey @oscargus I have been trying a bit now since couple of hours but I'm not sure does the bin test for test_sums_products.py give me an assertion error , locally on my system I get :
```
>>> Sum(x**bernoulli(4), (x, 1, 1000)).doit()
harmonic(1000, 1/30)
>>> Sum(1/sqrt(x), (x, 1, 1000)).doit()
harmonic(1000, 1/2)
>>> Sum(1/sqrt(x) , (x, 1, 10)).doit()
sqrt(10)/10 + sqrt(7)/7 + sqrt(6)/6 + sqrt(5)/5 + sqrt(3)/3 + 3*sqrt(2)/4 + 11/6
```
But no matter how I try to mold these I am getting an assertion error whenever I run the mandatory bin/test concrete as I have been trying to commit . Some examples 
```
    assert Sum(1/sqrt(x), (x, 1, 10)).doit() == sqrt(10)/10 + sqrt(7)/7 + sqrt(6)/6 + sqrt(5)/5 + sqrt(3)/3 + 3*sqrt(2)/4 + 11/6
AssertionError
    assert Sum(1/sqrt(x), (x, 1, 10)).doit() == Sum(1/sqrt(x), (x, 1, 10))
AssertionError
    assert Sum(x**bernoulli(4), (x, 1, 1000)).doit() == harmonic(1000, 1/30)
AssertionError
    assert Sum(x**bernoulli(6), (x, 1, 10)).doit() == 1 + 2**(1/42) + 3**(1/42) + 2**(1/21) + 5**(1/42) + 6**(1/42) + 7**(1/42) + 2**(1/14) + 3**(1/21) + 10**(1/42)
AssertionError
    assert Sum(1/sqrt(h), (h, 1, n)).doit() == harmonic(n, 1/2)
AssertionError
```
No matter how I'm trying to get through i'm not able too . Btw this doesn't trouble me much 
```
assert Sum(x**bernoulli(4), (x, 1, oo)).doit() == oo
assert Sum(x**bernoulli(5), (x, 1, 1000)).doit() == 1000
```
No idea what might be going wrong , will try to figure it out . I thought I was close ...... but may be not .Anyways the cases/above issues which I specifically wanted to fix , deal with `doit()` with upper bound as `oo ` and atleast those work really great as of now !

Reply:
Not really sure what the problem is, but `bernoulli(5) = 0`, so not sure why you test it like that. My comment was about testing a series which returns `bernoulli`, like `Sum(x**(S(9)/10), (x, 1, n)).doit()`